content/posts/irods-development-update-january-2024.md

```diff
@@ -0,0 +1,270 @@
+Title: iRODS Development Update: January 2024
+Date: 2024-01-15 10:00
+Author: Kory Draughn
+Slug: irods-development-update-january-2024
+Status: published
+
+
+Happy New Year!
+
+I hope everyone had a nice relaxing break. It's time for our first development update so let's get started.
+
+Since the release of iRODS 4.3.1, we've shifted focus to projects around the iRODS server.
+
+The first update is about externals packages. They are receiving a much needed improvement in the form of better version numbers. If you'd like to read more about that, see [https://github.com/irods/externals/issues/219](https://github.com/irods/externals/issues/219). Not only that, but work to provide better dependency declarations for externals packages is also in progress.
+
+Next up is the Metadata Guard rule engine plugin. We're now updating the plugin to understand atomic metadata operations. This will bring the plugin closer to covering all paths to metadata manipulation.
+
+We've fixed a few issues in the S3 resource plugin. Decoupled mode does a better job at honoring server redirects. Support for appending data has been improved as well. Due to **libs3** being absorbed into the plugin, we've re-licensed the plugin to **LGPLv3/GPLv2**. For those who are worried about the change in licensing, you can rest easy knowing you don't have to change anything you're doing.
+
+Storage Tiering 4.3.1.1 is now available. This latest release included several improvements, two of which were contributed by SURF _(a Consortium member)_. Please see the Technology Working Group notes below for details.
+
+A much needed tweak to the Python rule engine plugin is almost ready. To be specific, the next release of the plugin will make `global_vars` available to all imported modules. This means administrators can keep their **core.py** file clean and delegate the business logic to imported modules.
+
+Last month, we announced the release of the NetCDF project for iRODS 4.3.1. We're now looking into **possibly** archiving this project because we feel there are better alternatives (e.g. Python). If you have questions about this, don't hestitate to reach out to us via email or any other channel.
+
+That wraps up packaging and plugins. Let's talk about clients.
+
+First up is the Python iRODS client (PRC). For those users working in a 4.3.1 environment, you'll be happy to know support for **pam_password** has been implemented. Improvements for TTL have also been implemented. A new release of the PRC is inching closer and closer, so stay tuned.
+
+The HTTP API is seeing steady improvements. Support for HTTPS for OpenID Connect is very close. There's been a number of general improvements across the board. For example, configuring OpenID Connect is becoming simpler, the data objects endpoing now supports streaming reads like the C++ REST API, and the server now validates the configuration on startup.
+
```

Comment:
need a mention of k8s testing?

Reply:
Added some words. Not sure how I feel about them yet, but let me know what you think.